lib/src/migrator.dart

```diff
@@ -0,0 +1,247 @@
+    visitInterpolation(node.name);
+    visitArgumentInvocation(node.arguments);
+    if (node.name.asPlain == null) return;
+    var name = node.name.asPlain;
+    if (_currentMigration.localScope?.isLocalFunction(name) ?? false) return;
+    var namespace;
+    if (_currentMigration.functions.containsKey(name)) {
+      namespace =
+          _currentMigration.namespaceForNode(_currentMigration.functions[name]);
+    }
+    if (namespace == null) {
+      if (!builtInFunctionModules.containsKey(name)) return;
+      namespace = builtInFunctionModules[name];
+      name = builtInFunctionNameChanges[name] ?? name;
+      if (!_currentMigration.additionalUseRules.contains("sass:$namespace")) {
+        _currentMigration.additionalUseRules.add("sass:$namespace");
+      }
+    }
+    _currentMigration.patches.add(Patch(node.name.span, "$namespace.$name"));
+  }
+
+  /// Declares the function within the current scope before visiting it.
+  @override
+  void visitFunctionRule(FunctionRule node) {
+    _currentMigration.declareFunction(node);
+    super.visitFunctionRule(node);
+  }
+
+  /// Migrates @import to @use after migrating the imported file.
+  void visitImportRule(ImportRule node) {
+    if (node.imports.first is StaticImport) {
+      super.visitImportRule(node);
+      return;
+    }
+    if (node.imports.length > 1) {
+      throw UnimplementedError(
+          "Migration of @import rule with multiple imports not supported.");
+    }
+    var import = node.imports.first as DynamicImport;
+
+    if (_currentMigration.localScope != null) {
+      // TODO(jathak): Handle nested imports
+      return;
+    }
+    // TODO(jathak): Confirm that this import appears before other rules
+
+    var importMigration = _migrateStylesheet(import.url);
+
+    var overrides = [];
+    for (var variable in importMigration.configurableVariables) {
+      if (_currentMigration.variables.containsKey(variable)) {
+        var declaration = _currentMigration.variables[variable];
+        if (_currentMigration.namespaceForNode(declaration) == null) {
+          overrides.add("\$${declaration.name}: ${declaration.expression}");
+        }
+        // TODO(jathak): Remove this declaration from the current stylesheet if
+        //   it's not referenced before this point.
+      }
+    }
+    var config = "";
+    if (overrides.isNotEmpty) {
+      config = " with (\n  " + overrides.join(',\n  ') + "\n)";
+    }
+    _currentMigration.patches
+        .add(Patch(node.span, '@use ${import.span.text}$config'));
+    _currentMigration.namespaces[importMigration.path] =
+        namespaceForPath(import.url);
+
+    // Ensure that references to members transient dependencies can be
+    // namespaced.
+    _currentMigration.variables.addEntries(importMigration.variables.entries);
+    _currentMigration.mixins.addEntries(importMigration.mixins.entries);
+    _currentMigration.functions.addEntries(importMigration.functions.entries);
```

Comment:
Are these necessary to track now that you're also tracking the global scope? In fact, won't they be misleading, since once the current stylesheet is migrated to `@use` it *won't* make these members transitively visible?

Reply:
By adding the global members from imports to the current stylesheet, it lets us find existing references to transitively-visible members that need to be namespaced with the transitive dependency, not the direct one. This allows `namespaceForNode` to return the correct namespace (adding it if it hasn't been `@use`-d) by looking at the source path of the `VariableDeclaration`, `MixinRule`, or `FunctionRule` in one of these dictionaries.

Reply:
Oh, because the scope is bound to a `StylesheetMigration`? Maybe it should be a member of `_Migrator` instead... after all, the same global scope is shared across all stylesheets.

Reply:
I don't think a single global scope would work when the migrator is running on multiple entrypoints. The scope would need to be cleared between each entrypoint, but the stored global members for each dependency need to remain available in case a future entrypoint also re-uses them.

Reply:
I suspect that you'll end up needing to abandon shared state across entrypoints. The way names are resolved in an individual file is highly dependent on the global state at the time it's loaded, so I suspect you'll run into lots of nasty edge cases if you don't treat each entrypoint as an entirely separate run.

Reply:
Oh, okay. That makes sense. I'll change it so the scoping information is just stored in `_Migrator`.

What should I do in the case where the user wants to migrate two entrypoints that share a dependency, but the dependency would be migrated differently depending on which entrypoint is loading it?

Reply:
Probably just throw an error.